### PR TITLE
8256213: Remove os::split_reserved_memory

### PR DESCRIPTION
**Pull request by yminqi**
Hi, Please review 
  This is the second step for removing problematic os::split_reserved_memory on Windows. The first step has been accomplished under bug 8255917 (https://bugs.openjdk.java.net/browse/JDK-8255917).
  This step remove the function split_reserved_memory which no longer be called after first step.

Tests: tier1-4

Thanks
Yumin

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8256213](https://bugs.openjdk.java.net/browse/JDK-8256213): Remove os::split_reserved_memory

### Reviewers
 * [Calvin Cheung](https://openjdk.java.net/census#ccheung) (@calvinccheung - **Reviewer**) ⚠️ Review applies to 4318c9a3678c7a6b89bd221e323b387b4ebc767c
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1825/head:pull/1825`
`$ git checkout pull/1825`

### CODE REVIEW

File: src/hotspot/share/memory/virtualspace.hpp [hunk 1]
```diff
@@ -76,17 +76,11 @@ class ReservedSpace {
...
-  //  Note that this may cause the original space to loose its content.
-  // If split==false, the resulting space will be just a hotspot-internal representation
-  //  of a sub section of the underlying mapping.
-  ReservedSpace first_part(size_t partition_size, size_t alignment, bool split = false);
+  ReservedSpace first_part(size_t partition_size, size_t alignment);
   ReservedSpace last_part (size_t partition_size, size_t alignment);
 
   // These simply call the above using the default alignment.
-  inline ReservedSpace first_part(size_t partition_size, bool split = false);
+  inline ReservedSpace first_part(size_t partition_size/*, bool split = false*/);
```
Comment (calvinccheung) [inline src/hotspot/share/memory/virtualspace.hpp:83]: Forgot to remove the comment code `/*, bool split = false*/`?

**Review by calvinccheung — APPROVED** [review]
Looks good other than a minor cleanup.

File: src/hotspot/share/memory/virtualspace.hpp (hunk 1 above)
Comment (yminqi) [reply src/hotspot/share/memory/virtualspace.hpp:83]: Yes, thanks for the review.

**Review by iklam — APPROVED** [review]
LGTM

**Comment by yminqi** [comment]
> LGTM

Thanks for review!

**Comment by tstuefe** [comment]
Nice to see this gone ! 

..Thomas

**Pull request closed without merging**